src/renderer/base/renderer.cpp

```diff
@@ -1304,16 +1281,9 @@ void Renderer::_ScrollPreviousSelection(const til::point delta)
 {
     if (delta != til::point{ 0, 0 })
     {
-        for (auto& sr : _previousSelection)
+        for (auto& rc : _previousSelection)
         {
-            // Get a rectangle representing this piece of the selection.
-            til::rect rc{ Viewport::FromExclusive(sr).ToInclusive() };
-
-            // Offset the entire existing rectangle by the delta.
             rc += delta;
-
-            // Store it back into the vector.
-            sr = Viewport::FromInclusive(rc.to_small_rect()).ToExclusive();
         }

```

Comment:
i just have to trust that this is right lol

Reply:
This one is double triple checked! It was easy to prove that this change is correct, because the `FromInclusive`/`FromExclusive` and `ToInclusive`/`ToExclusive` functions are now strict about the parameter types they take or return (inclusive/exclusive rect). Afterwards the calls could be safely removed.